src/Special/WSpsSpecial.php

```diff
@@ -533,11 +533,11 @@ public function execute( $sub ) {
 				);
 				// First check if we have SMW
 				$specialSMW = new PSSpecialSMWQeury();
+				$SMWQueryDisabledText = '';
 				if ( !$specialSMW->isExtensionInstalled( 'SemanticMediaWiki' ) ) {
-					$out->addHTML( self::makeAlert( wfMessage( 'wsps-special_custom_query_we_need_smw' )->text() ) );
-					$out->addHTML( $style );
-
-					return true;
+					$SMWQueryDisabledText = self::makeAlert(
```

Comment:
Watch out! You may still try to execute SMW-related code here (Which will probably throw an exception of the form "class not found" or something?)

You can better try to not execute SMW-related things when SMW is not enabled.

Also, please, (maybe not in this pull request), refactor this function: Split it into multiple functions that have descriptive names, such that it is not one big atrocious function but several small nice functions.

Reply:
No SMW code is executed when SMW is not installed. Since no SMW query request code is ever rendered if it is not installed. 
And I agree with the refactoring.. is planned.